internal/client/client.go

```diff
@@ -9,91 +10,68 @@ import (
 	"github.com/ionos-cloud/ionosctl/v6/pkg/config"
 
 	"github.com/ionos-cloud/ionosctl/v6/pkg/constants"
-	sdkgoauth "github.com/ionos-cloud/sdk-go-auth"
-	certmanager "github.com/ionos-cloud/sdk-go-cert-manager"
-	registry "github.com/ionos-cloud/sdk-go-container-registry"
-	dataplatform "github.com/ionos-cloud/sdk-go-dataplatform"
-	mongo "github.com/ionos-cloud/sdk-go-dbaas-mongo"
-	postgres "github.com/ionos-cloud/sdk-go-dbaas-postgres"
-	dns "github.com/ionos-cloud/sdk-go-dns"
-	cloudv6 "github.com/ionos-cloud/sdk-go/v6"
 	"github.com/spf13/viper"
 )
 
-type Client struct {
-	CloudClient        *cloudv6.APIClient
-	AuthClient         *sdkgoauth.APIClient
-	CertManagerClient  *certmanager.APIClient
-	PostgresClient     *postgres.APIClient
-	MongoClient        *mongo.APIClient
-	DataplatformClient *dataplatform.APIClient
-	RegistryClient     *registry.APIClient
-	DnsClient          *dns.APIClient
-}
-
-func appendUserAgent(userAgent string) string {
-	return fmt.Sprintf("%v_%v", viper.GetString(constants.CLIHttpUserAgent), userAgent)
-}
+var once sync.Once
+var instance *Client
 
-func newClient(name, pwd, token, hostUrl string) (*Client, error) {
-	if token == "" && (name == "" || pwd == "") {
-		return nil, errors.New("username, password or token incorrect")
+func selectAuthLayer(layers []Layer) (values map[string]string, usedLayer Layer, err error) {
+	for _, layer := range layers {
+		token := viper.GetString(layer.TokenKey)
+		username := viper.GetString(layer.UsernameKey)
+		password := viper.GetString(layer.PasswordKey)
+
+		// TODO (Ask the team for feedback):
+		// If token is set and invalid, we could still try the user & pass here, and choose them over the invalid token.
+		// However this could be inconsistent with the Ansible / TF approach and could confuse the user.
+		// TODO END
+		if layer.TokenKey != "" && token != "" ||
```

Comment:
> do we need to check layer.TokenKey != "" ?

I was thinking that a valid use case of this new layered approach would be to define a layer with only `Token` or only `Username & Password support, in which case the unsupported key for the layer would be empty. 

That being said, yeah, these checks are redundant because `viper.GetString("")` would still return an empty string - so I see your point and we can totally remove these checks.
 
> Can't you just check err above for GetString and only set token if err!= nil? This way here you can just check token?

I'm not entirely sure what you mean here, `GetString` doesn't return an error, but yeah, we can skip the layer.<Token/User>Key checks.